packages/flutter/lib/src/material/slider.dart

```diff
@@ -643,14 +639,12 @@ class _RenderSlider extends RenderBox {
       _state.overlayController.forward();
       if (showValueIndicator) {
         _state.valueIndicatorController.forward();
-        if (_state.interactionTimer.isActive) {
-          _state.interactionTimer.cancel();
-        }
+        _state.interactionTimer?.cancel();
         _state.interactionTimer = new Timer(_minimumInteractionTime * timeDilation, () {
           if (!_active && _state.valueIndicatorController.status == AnimationStatus.completed) {
             _state.valueIndicatorController.reverse();
           }
-          _state.interactionTimer.cancel();
+          _state.interactionTimer?.cancel();
```

Comment:
this line doesn't make sense. You know it's not null and you know it doesn't need canceling, at this point.

I would just replace this line with setting the timer to null.

Reply:
I don't know that it's not null: the timer could have been set to null somewhere else before the callback gets called.

I see what you mean about knowing that it doesn't need canceling, though.